diff --git a/desktop/libs/notebook/src/notebook/connectors/base.py b/desktop/libs/notebook/src/notebook/connectors/base.py
index 2d8c0e1f2f..79741911e6 100644
--- a/desktop/libs/notebook/src/notebook/connectors/base.py
+++ b/desktop/libs/notebook/src/notebook/connectors/base.py
@@ -486,6 +486,9 @@ def get_api(request, snippet):
     elif interpreter['options'] and interpreter['options'].get('url', '').find('presto') >= 0:
       from notebook.connectors.jdbc_presto import JdbcApiPresto
       return JdbcApiPresto(request.user, interpreter=interpreter)
+    elif interpreter['options'] and interpreter['options'].get('url', '').find('trino') >= 0:
+      from notebook.connectors.jdbc_trino import JdbcApiTrino
+      return JdbcApiTrino(request.user, interpreter=interpreter)
     elif interpreter['options'] and interpreter['options'].get('url', '').find('clickhouse') >= 0:
       from notebook.connectors.jdbc_clickhouse import JdbcApiClickhouse
       return JdbcApiClickhouse(request.user, interpreter=interpreter)
